### PR TITLE
TF-2684 Support copy text in blue bar

### PR DESCRIPTION
**Pull request by dab246**
## Issue

#2684 

## Resolved

- Web:

https://github.com/linagora/tmail-flutter/assets/80730648/407e7114-0674-49b9-969a-ac08a94f0d5e

- Mobile

https://github.com/linagora/tmail-flutter/assets/80730648/54c4af45-0a13-4e4c-a0a5-f8ba770c9d15

### CODE REVIEW

**Review by tddang-linagora — APPROVED** [review]

File: core/lib/presentation/action/action_callback_define.dart [hunk 1]
```diff
@@ -2,4 +2,5 @@
 import 'package:flutter/material.dart';
 
 typedef OnTapActionCallback = void Function();
-typedef OnTapActionAtPositionCallback = void Function(RelativeRect position);
\ No newline at end of file
+typedef OnTapActionAtPositionCallback = void Function(RelativeRect position);
+typedef OnLongPressActionCallback = void Function();
```
Comment (hoangdat) [inline core/lib/presentation/action/action_callback_define.dart:6]: remove

**Comment by hoangdat** [comment]
some time very hard to select text, and found the console log

```
AppLogger::initLogger::runZonedGuarded:FlutterError.onError: ../canvaskit/chromium/canvaskit.js 115:52                                            getGlyphPositionAtCoordinate../lib/_engine/engine/canvaskit/canvaskit_api.dart 3263:8                            SkParagraphExtension.getGlyphPositionAtCoordinate../lib/_engine/engine/canvaskit/text.dart 651:60                                     getPositionForOffset../packages/flutter/src/painting/text_painter.dart 1489:24                           getPositionForOffset../packages/flutter/src/rendering/paragraph.dart 976:25                              getPositionForOffset../packages/flutter/src/rendering/paragraph.dart 1477:64                             [_updateSelectionEdge]../packages/flutter/src/rendering/paragraph.dart 1409:22                             dispatchSelectionEvent../packages/flutter/src/widgets/scrollable.dart 1482:17                              ensureChildUpdated../packages/flutter/src/widgets/scrollable.dart 1442:9                               dispatchSelectionEventToChild../packages/flutter/src/widgets/selectable_region.dart 2461:33                       [_adjustSelection]../packages/flutter/src/widgets/selectable_region.dart 2281:85                       handleSelectionEdgeUpdate../packages/flutter/src/widgets/scrollable.dart 1225:42                              handleSelectionEdgeUpdate../packages/flutter/src/widgets/selectable_region.dart 2298:18                       dispatchSelectionEvent../packages/flutter/src/widgets/selection_container.dart 183:19                      dispatchSelectionEvent../packages/flutter/src/widgets/selectable_region.dart 2346:22                       dispatchSelectionEventToChild../packages/flutter/src/widgets/scrollable.dart 1459:18                              dispatchSelectionEventToChild../packages/flutter/src/widgets/selectable_region.dart 2461:33                       [_adjustSelection]../packages/flutter/src/widgets/selectable_region.dart 2281:85                       handleSelectionEdgeUpdate../packages/flutter/src/widgets/scrollable.dart 1225:42                              handleSelectionEdgeUpdate../packages/flutter/src/widgets/selectable_region.dart 2298:18                       dispatchSelectionEvent../packages/flutter/src/widgets/selection_container.dart 183:19                      dispatchSelectionEvent../packages/flutter/src/widgets/selectable_region.dart 2346:22                       dispatchSelectionEventToChild../packages/flutter/src/widgets/selectable_region.dart 1576:18                       dispatchSelectionEventToChild../packages/flutter/src/widgets/selectable_region.dart 2461:33                       [_adjustSelection]../packages/flutter/src/widgets/selectable_region.dart 2281:85                       handleSelectionEdgeUpdate../packages/flutter/src/widgets/selectable_region.dart 1545:18                       handleSelectionEdgeUpdate../packages/flutter/src/widgets/selectable_region.dart 2298:18                       dispatchSelectionEvent../packages/flutter/src/widgets/selection_container.dart 183:19                      dispatchSelectionEvent../packages/flutter/src/widgets/selectable_region.dart 1002:20                       [_selectStartTo]../packages/flutter/src/widgets/selectable_region.dart 523:9                         [_handleMouseDragStart]../packages/flutter/src/gestures/tap_and_drag.dart 1217:49                           <fn>../packages/flutter/src/gestures/recognizer.dart 275:24                              invokeCallback../packages/flutter/src/gestures/tap_and_drag.dart 1217:7                            [_checkDragStart]../packages/flutter/src/gestures/tap_and_drag.dart 1127:5                            [_acceptDrag]../packages/flutter/src/gestures/tap_and_drag.dart 992:7                             acceptGesture../packages/flutter/src/gestures/arena.dart 281:11                                   [_resolveInFavorOf]../packages/flutter/src/gestures/arena.dart 239:9                                    [_resolve]../packages/flutter/src/gestures/arena.dart 53:5                                     resolve../packages/flutter/src/gestures/recognizer.dart 375:12                              resolve../packages/flutter/src/gestures/tap_and_drag.dart 1161:9                            [_checkDrag]../packages/flutter/src/gestures/tap_and_drag.dart 1074:11                           handleEvent../packages/flutter/src/gestures/pointer_router.dart 98:7                            [_dispatch]../packages/flutter/src/gestures/pointer_router.dart 143:9                           <fn>../dart-sdk/lib/_internal/js_dev_runtime/private/linked_hash_map.dart 21:7           forEach../packages/flutter/src/gestures/pointer_router.dart 141:17                          [_dispatchEventToRoutes]../packages/flutter/src/gestures/pointer_router.dart 127:7                           route../packages/flutter/src/gestures/binding.dart 488:5                                  handleEvent../packages/flutter/src/gestures/binding.dart 468:14                                 dispatchEvent../packages/flutter/src/rendering/binding.dart 439:11                                dispatchEvent../packages/flutter/src/gestures/binding.dart 413:7                                  [_handlePointerEventImmediately]../packages/flutter/src/gestures/binding.dart 376:5                                  handlePointerEvent../packages/flutter/src/gestures/binding.dart 323:7                                  [_flushPointerEventQueue]../packages/flutter/src/gestures/binding.dart 292:9                                  [_handlePointerDataPacket]../dart-sdk/lib/async/zone.dart 1415:13                                              _rootRunUnary../dart-sdk/lib/async/zone.dart 1308:19                                              runUnary../dart-sdk/lib/async/zone.dart 1217:7                                               runUnaryGuarded../lib/_engine/engine/platform_dispatcher.dart 1291:9                                invoke1../lib/_engine/engine/platform_dispatcher.dart 273:5                                 invokeOnPointerDataPacket../lib/_engine/engine/pointer_binding.dart 168:30                                    [_onPointerData]../lib/_engine/engine/pointer_binding.dart 768:16                                    <fn>../lib/_engine/engine/pointer_binding.dart 720:7                                     <fn>../lib/_engine/engine/pointer_binding.dart 317:9                                     loggedHandler../dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 574:37  _checkAndCall../dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 579:39  dcall
```

**Comment by hoangdat** [comment]
Hi @dab246 
I can not get the log again, but still hard in select text in blue bar, especially in tablet break point

https://github.com/linagora/tmail-flutter/assets/6462404/8e20f8df-2aa8-401b-bd5a-896081d0b2fe

**Comment by dab246** [comment]
> hard in select text in blue bar,

 @hoangdat I fixed it. Please see video demo below and review again. Thank.

https://github.com/linagora/tmail-flutter/assets/80730648/34b8db97-85ed-4190-9744-c1bcc7c678d8

**Review by hoangdat — APPROVED** [review]

**Pull request merged**